### DISCUSSION TITLE
Tasmota Source 13.2.0.1 from arendst don´t compile ES32 Firmwareimage

### DISCUSSION THREAD

**Discussion by HofeBY** [arendst/Tasmota#19813] [Support request]
Compiling ESP32 Firmware under Tasmota 13.2.0.1 is not working !

Compiling ESP32 Firmware under Tasmota 13.1.0.4 is working !

I can´t compile ESP32 FirmwareIage under Tasmota 13.2.0.1 !
Windows10 X64
Visual Stutio Code 1.183.1
PlatformIO IDE v3.3.1
reloading the source brings the same error.

Here my compilerlog:

*  Executing task: C:\Users\Station\.platformio\penv\Scripts\platformio.exe run --environment tasmota32 

Processing tasmota32 (framework: arduino; platform: https://github.com/tasmota/platform-espressif32/releases/download/2023.10.04/platform-espressif32.zip; board: esp32)
---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------Verbose mode can be enabled via `-v, --verbose` option
-DCONFIG_TASMOTA_FLASHMODE_DIO
*** use provided user_config_override.h as planned ***
*** use provided platformio_override.ini as planned ***
*** use provided platformio_tasmota_cenv.ini as planned ***     
*** use provided platformio_tasmota_core3_env.ini as planned ***
CONFIGURATION: https://docs.platformio.org/page/boards/espressif32/esp32.html
PLATFORM: Espressif 32 (2023.10.4) > Espressif Generic ESP32 >= 4M Flash PSRAM, Tasmota 2880k Code/OTA, 320k FS
HARDWARE: ESP32 160MHz, 320KB RAM, 4MB Flash
DEBUG: Current (cmsis-dap) External (cmsis-dap, esp-bridge, esp-prog, iot-bus-jtag, jlink, minimodule, olimex-arm-usb-ocd, olimex-arm-usb-ocd-h, olimex-arm-usb-tiny-h, olimex-jtag-tiny, tumpa)
PACKAGES:
 - framework-arduinoespressif32 @ 2.0.14
 - tool-esptoolpy @ 1.40700.0+dev (4.7.0)
 - tool-mklittlefs @ 3.2.0
 - tool-mkspiffs @ 2.230.0 (2.30)
 - toolchain-xtensa-esp32 @ 8.4.0+2021r2-patch5
Converting tasmota.ino
LDF: Library Dependency Finder -> https://bit.ly/configure-pio-ldf
LDF Modes: Finder ~ chain, Compatibility ~ strict
Found 128 compatible libraries
Scanning dependencies...
Dependency Graph
|-- Adafruit CCS811 Library @ 1.0.0        
|-- Adafruit LittlevGL Glue Library @ 2.0.0
|-- Adafruit MAX31865 library @ 1.0.1      
|-- Adafruit MCP9808 Library @ 1.1.2       
|-- Adafruit PM25 AQI Sensor @ 1.0.6       
|-- Adafruit seesaw Library @ 1.3.1        
|-- Adafruit SGP30 Sensor @ 1.2.0
|-- Adafruit SGP40 Sensor @ 1.1.0
|-- Adafruit VEML7700 Library @ 1.0.0
|-- ESP8266Audio @ 1.9.5
|-- Berry @ 7.0
|-- Berry Matter protocol implementation @ 0.1
|-- Berry Tasmota mapping @ 1.0
|-- Berry int64 implementation for 32 bits architecture @ 1.0
|-- Berry mapping to C @ 1.0
|-- BME8563
|-- headers
|-- ccronexpr
|-- WebServer @ 2.0.0
|-- EEPROM 24C128_256_521
|-- ESP32-to-ESP8266-compat @ 0.0.2
|-- FFat @ 2.0.0
|-- FS @ 2.0.0
|-- ams @ 1.0
|-- HttpClient light @ 1.0
|-- I2Cdevlib-Core
|-- LibTeleinfo @ 1.1.5
|-- lvgl @ 8.3.10
|-- Berry mapping to LVGL @ 1.0
|-- arduino-mcp2515-1.0.1
|-- MPU6050 I2C
|-- Mutichannel_Gas_Sensor @ 0.0.1
|-- NimBLE-Arduino @ 1.4.1
|-- NTP Library
|-- OpenTherm Library @ 0.9.0
|-- Process_control @ 1.0.0
|-- re1.5 @ 0.9
|-- ScioSense ENS16x @ 8.0.0
|-- ScioSense ENS210 @ 3.0.0
|-- Sensirion I2C SGP41 @ 0.1.0
|-- SevenSegmentTM1637 @ 1.1.0
|-- ESP Mail Client @ 3.4.9
|-- LVGL_assets @ 7.0
|-- UdpListener @ 1.0
|-- Unishox Compressor Decompressor highly customized and optimized for ESP8266 and Tasmota @ 1.0
|-- VL53L0X @ 1.0.2
|-- VL53L1X @ 1.0.1
|-- TLS mini client derived from Arduino @ 1.0
|-- Zip-readonly-FS @ 1.0
|-- A4988_Stepper @ 0.0.1
|-- Adafruit Fingerprint Sensor Library @ 2.1.0
|-- Adafruit GFX Library @ 1.5.6
|-- Adafruit LED Backpack Library @ 1.1.6
|-- Adafruit SH1106-gemu-1.0 @ 1.0
|-- Adafruit SSD1306 @ 1.3.0
|-- Adafruit SSD1331 OLED Driver Library for Arduino @ 1.2.0
|-- Adafruit TSL2591
|-- Arduino ST7789 Library @ 0.9.5
|-- TTGO TWatch Library @ 1.2.0
|-- base64 @ 1.1.1
|-- BME68x
|-- Micro-RTSP @ 0.1.6
|-- DHT sensor library @ 1.4.3
|-- DNSServer @ 2.0.0
|-- DS2480 Library @ 0.1.0
|-- Waveshare esp 2.9 inch e-paper display driver @ 1.0
|-- Waveshare esp 4.2 inch e-paper display driver @ 1.0
|-- Display renderer @ 1.0
|-- ES7210 @ 1.0.0
|-- ES7243e @ 1.0.0
|-- ES8156 @ 1.0.0
|-- ES8311 @ 1.0.0
|-- ESP KNX IP Library @ 0.5.2
|-- ESP8266SAM @ 1.0.1
|-- Ethernet @ 2.0.0
|-- Ext-printf @ 1.0
|-- FrogmoreScd30
|-- FrogmoreScd40
|-- FT5206_Library @ 1.0.0
|-- GT911
|-- HPMA115S0 Arduino Library @ 1.0.0
|-- ILI9341 @ 1.0.0
|-- IRremoteESP8266 @ 2.8.6
|-- JSMN JSON parser customized and optimized for ESP8266 and Tasmota @ 1.0
|-- KeeloqLib @ 1.1
|-- mp3_shine_esp32 @ 1.0.0
|-- LedControl @ 1.0.6
|-- LinkedList
|-- LiquidCrystal_I2C
|-- LittleFS @ 2.0.0
|-- TasmotaLList @ 1.0
|-- LOLIN_HP303B @ 1.0.0
|-- MFRC522 @ 1.4.7
|-- MLX90640
|-- MPU_accel
|-- NeoPixelBus @ 2.6.7
|-- NewPing @ 1.9.1
|-- OneWire @ 2.3.2
|-- PubSubClient @ 2.8
|-- RA8876 @ 1.0.2
|-- rc-switch @ 1.0.0
|-- RF24 @ 1.3.3
|-- SD @ 2.0.0
|-- SD_MMC @ 2.0.0
|-- Sensirion I2C SEN5X @ 0.2.0
|-- SPI @ 2.0.0
|-- SSD3115 @ 1.0
|-- TasmotaModbus @ 3.6.0
|-- TasmotaSerial @ 3.6.0
|-- Ticker @ 2.0.0
|-- TM1638plus @ 1.7.0
|-- Joba_Tsl2561 @ 2.0.10
|-- BearSSL @ 0.6
|-- universal display Library @ 0.1
|-- WiFi @ 2.0.0
|-- Wire @ 2.0.0
|-- WM8960 @ 1.0.0
|-- XPT2046_Touchscreen
Building in release mode
Compiling .pio\build\tasmota32\src\tasmota.ino.cpp.o
Compiling .pio\build\tasmota32\src\tasmota_support\homekit.c.o
Building .pio\build\tasmota32\bootloader.bin
Generating partitions .pio\build\tasmota32\partitions.bin
esptool.py v4.7-dev
Creating esp32 image...
Merged 1 ELF section
Successfully created esp32 image.
Compiling .pio\build\tasmota32\libc4c\Wire\Wire.cpp.o
Compiling .pio\build\tasmota32\libfc2\Adafruit_CCS811-1.0.0.14\Adafruit_CCS811.cpp.o
In file included from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal.h:83,
                 from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:36,
                 from C:\Users\Station\AppData\Local\Temp\tmpfdcstaiw:1:
C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.h:29:10: fatal error: pins_arduino.h: No such file or directory

**********************************************************************
* Looking for pins_arduino.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:pins_arduino.h"
* Web  > https://registry.platformio.org/search?q=header:pins_arduino.h
*
**********************************************************************

 #include "pins_arduino.h"
          ^~~~~~~~~~~~~~~~
compilation terminated.
In file included from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal.h:83,
                 from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/libraries/Wire/src/Wire.h:29,
                 from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/libraries/Wire/src/Wire.cpp:34:
C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.h:29:10: fatal error: pins_arduino.h: No such file or directory

**********************************************************************
* Looking for pins_arduino.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:pins_arduino.h"
* Web  > https://registry.platformio.org/search?q=header:pins_arduino.h
*
**********************************************************************

 #include "pins_arduino.h"
          ^~~~~~~~~~~~~~~~
compilation terminated.
Compiling .pio\build\tasmota32\libdf0\FS\FS.cpp.o
Compiling .pio\build\tasmota32\libdf0\FS\vfs_api.cpp.o
In file included from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal.h:83,
                 from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:36,
                 from lib/lib_i2c/Adafruit_CCS811-1.0.0.14/Adafruit_CCS811.h:5,
                 from lib/lib_i2c/Adafruit_CCS811-1.0.0.14/Adafruit_CCS811.cpp:1:
C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.h:29:10: fatal error: pins_arduino.h: No such file or directory

**********************************************************************
* Looking for pins_arduino.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:pins_arduino.h"
* Web  > https://registry.platformio.org/search?q=header:pins_arduino.h
*
**********************************************************************

 #include "pins_arduino.h"
          ^~~~~~~~~~~~~~~~
compilation terminated.
Compiling .pio\build\tasmota32\libfff\freetype\base\ftbase.c.o
Compiling .pio\build\tasmota32\libfff\freetype\base\ftbbox.c.o
Compiling .pio\build\tasmota32\libfff\freetype\base\ftbdf.c.o
Compiling .pio\build\tasmota32\libfff\freetype\base\ftbitmap.c.o
*** [.pio\build\tasmota32\src\tasmota.ino.cpp.o] Error 1
*** [.pio\build\tasmota32\libc4c\Wire\Wire.cpp.o] Error 1
*** [.pio\build\tasmota32\libfc2\Adafruit_CCS811-1.0.0.14\Adafruit_CCS811.cpp.o] Error 1
In file included from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal.h:83,
                 from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:36,
                 from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/libraries/FS/src/FS.h:25,
                 from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/libraries/FS/src/FS.cpp:21:
C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.h:29:10: fatal error: pins_arduino.h: No such file or directory

**********************************************************************
* Looking for pins_arduino.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:pins_arduino.h"
* Web  > https://registry.platformio.org/search?q=header:pins_arduino.h
*
**********************************************************************

 #include "pins_arduino.h"
          ^~~~~~~~~~~~~~~~
compilation terminated.
In file included from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal.h:83,
                 from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/Arduino.h:36,
                 from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/libraries/FS/src/FS.h:25,
                 from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/libraries/FS/src/vfs_api.h:18,
                 from C:/Users/Station/.platformio/packages/framework-arduinoespressif32/libraries/FS/src/vfs_api.cpp:15:
C:/Users/Station/.platformio/packages/framework-arduinoespressif32/cores/esp32/esp32-hal-gpio.h:29:10: fatal error: pins_arduino.h: No such file or directory

**********************************************************************
* Looking for pins_arduino.h dependency? Check our library registry!
*
* CLI  > platformio lib search "header:pins_arduino.h"
* Web  > https://registry.platformio.org/search?q=header:pins_arduino.h
*
**********************************************************************

 #include "pins_arduino.h"
          ^~~~~~~~~~~~~~~~
compilation terminated.
*** [.pio\build\tasmota32\libdf0\FS\FS.cpp.o] Error 1
*** [.pio\build\tasmota32\libdf0\FS\vfs_api.cpp.o] Error 1
====================================================================================== [FAILED] Took 64.77 seconds ======================================================================================
Environment    Status    Duration
-------------  --------  ------------
tasmota32      FAILED    00:01:04.771
================================================================================= 1 failed, 0 succeeded in 00:01:04.771 =================================================================================
 *  The terminal process "C:\Users\Station\.platformio\penv\Scripts\platformio.exe 'run', '--environment', 'tasmota32'" terminated with exit code: 1. 
 *  Terminal will be reused by tasks, press any key to close it.

**Comment by Jason2866** [comment]
Maybe Platformio is borked. Close `VSC` and delete folder `c:/Users/Station/.platformio` with all content and all sub-folders. Delete in Tasmota folder `.pio` and `.cache` Start `VSC` Platformio will be reinstalled.

**Reply by HofeBY** [reply]
Around 1h later a new without changing anything here the new compiler run downloaded some new SW parts and compiled well.
Someone repaired the source. Well done and thx.

here last tail of compiler run:

Used static DRAM:   71144 bytes (  53436 remain, 57.1% used)
      .data size:   22048 bytes
      .bss  size:   49096 bytes
Used static IRAM:   94370 bytes (  36702 remain, 72.0% used)
      .text size:   93343 bytes
   .vectors size:    1027 bytes
Used Flash size : 1793299 bytes
           .text: 1180667 bytes
         .rodata:  612376 bytes
Total image size: 1909717 bytes (.bin may be padded larger)
================================================================================== [SUCCESS] Took 123.34 seconds ==================================================================================
Environment    Status    Duration
-------------  --------  ------------
tasmota32      SUCCESS   00:02:03.337
=================================================================================== 1 succeeded in 00:02:03.337 =================================================================================== *  Terminal will be reused by tasks, press any key to close it.
